### PR TITLE
[GLUTEN-4831][VL] Support StructType in HashAggregate

### PR DESCRIPTION
**Pull request by WangGuangxin**
## What changes were proposed in this pull request?

Support StructType in HashAggregate

(Fixes: \#4831)

### CODE REVIEW

File: gluten-core/src/main/scala/io/glutenproject/execution/HashAggregateExecBaseTransformer.scala [hunk 1]
```diff
@@ -99,6 +99,7 @@ abstract class HashAggregateExecBaseTransformer(
         true
       case _: NumericType => true
       case _: ArrayType => true
+      case s: StructType => true
```
Comment (PHILO-HE) [inline gluten-core/src/main/scala/io/glutenproject/execution/HashAggregateExecBaseTransformer.scala:102]: Nit: remove `s`.

**Review by PHILO-HE — COMMENTED** [review]
Please do a rebase and resolve conflicts. Thanks!

File: gluten-core/src/main/scala/io/glutenproject/execution/HashAggregateExecBaseTransformer.scala (hunk 1 above)
Comment (WangGuangxin) [reply gluten-core/src/main/scala/io/glutenproject/execution/HashAggregateExecBaseTransformer.scala]: fixed and rebased. thanks.

**Review by PHILO-HE — APPROVED** [review]
Looks good! cc @zhztheplayer

**Pull request merged**